azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java

```diff
@@ -1364,6 +1382,52 @@ public String handle(ResultSet rs) throws SQLException {
     }
   }
 
+  private static class FetchRecentlyFinishedFlows implements
+    ResultSetHandler<List<ExecutableFlow>> {
+    // Execution_flows table is already indexed by end_time
+    private static String FETCH_RECENTLY_FINISHED_FLOW =
+        "SELECT exec_id, enc_type, flow_data FROM execution_flows "
+            + "WHERE end_time > ? AND status IN (?, ?, ?)";
+
+    @Override
+    public List<ExecutableFlow> handle(
+        ResultSet rs) throws SQLException {
+      if (!rs.next()) {
+        return Collections.emptyList();
+      }
+
+      List<ExecutableFlow> execFlows = new ArrayList<>();
+      do {
+        int id = rs.getInt(1);
+        int encodingType = rs.getInt(2);
+        byte[] data = rs.getBytes(3);
+
+        if (data != null) {
+          EncodingType encType = EncodingType.fromInteger(encodingType);
+          Object flowObj;
+          try {
+            if (encType == EncodingType.GZIP) {
```

Comment:
this piece of code is duplicate all over the place, can we make it a util? e.x it's also used in https://github.com/azkaban/azkaban/pull/1079